lib/node_modules/@stdlib/math/base/complex/divide/lib/cdiv.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+// MODULES //
+
+var FLOAT64_BIGGEST = require( '@stdlib/math/constants/float64-max' );
+var FLOAT64_SMALLEST = require( '@stdlib/math/constants/float64-smallest-normal' );
+var EPS = require( '@stdlib/math/constants/float64-eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var max = require( '@stdlib/math/base/special/max' );
+var robustInternal = require( './robust_internal.js' );
+
+
+// VARIABLES //
+
+var LARGE_THRESHOLD = FLOAT64_BIGGEST * 0.5;
+var SMALL_THRESHOLD = FLOAT64_SMALLEST * ( 2.0 / EPS );
+var RECIP_EPS_SQR = 2.0 / ( EPS * EPS );
+
+
+// MAIN //
+
+/**
+* Divides two complex numbers. Allocates storage if not provided.
+*
+* ## References
+*
+* * Baudin, Michael, and Robert L. Smith. 2012. "A Robust Complex Division in Scilab" abs/1210.4539 (October). <https://arxiv.org/abs/1210.4539>. https://arxiv.org/pdf/1210.4539.pdf
+*
+* @param {(Array|TypedArray|Object)} [out] - optional output array
+* @param {number} re1 - real component
+* @param {number} im1 - imaginary component
+* @param {number} re2 - real component
+* @param {number} im2 - imaginary component
+* @returns {(Array|TypedArray|Object)} real and imaginary components
+*
+* @example
+* var v = divide( -13.0, -1.0, -2.0, 1.0 );
+* // returns [ 5.0, 3.0 ]
+*
+* @example
+* var out = [];
+*
+* var v = divide( out, -13.0, -1.0, -2.0, 1.0 );
+* // out => [ 5.0, 3.0 ]
```

Comment:
// returns [ 5.0, 3.0 ]

That the contents of `out` is the same as that of the return value can be inferred on line `46`.

In the above example, because the focus is on `out`, a reader is not provided any indication as to the value of `v`.